docs/developer-docs/setup/pulling-canister-dependencies.md

```diff
@@ -0,0 +1,430 @@
+    }
+  }
+}
+```
+
+### Step 4: Deploy the pulled dependencies on a local replica using the `dfx deps deploy` command.
+
+Running the `dfx deps deploy` command will:
+
+1. First, create the dependencies on the local replica with the same mainnet canister ID.
+2. Then, it will install the downloaded wasm with the init arguments in the `init.json` file.
+
+You can also specify the name or principal to deploy one particular dependency.
+
+Using the example above, we can run the following command to deploy all dependencies:
+
+```
+dfx deps deploy
+Creating canister: yofga-2qaaa-aaaaa-aabsq-cai
+Installing canister: yofga-2qaaa-aaaaa-aabsq-cai
+Creating canister: yhgn4-myaaa-aaaaa-aabta-cai (dep_b)
+Installing canister: yhgn4-myaaa-aaaaa-aabta-cai (dep_b)
+Creating canister: yahli-baaaa-aaaaa-aabtq-cai (dep_c)
+Installing canister: yahli-baaaa-aaaaa-aabtq-cai (dep_c)
+```
+
+To deploy one particular dependency, the following command can be used: 
+
+```
+dfx deps deploy yofga-2qaaa-aaaaa-aabsq-cai
+Installing canister: yofga-2qaaa-aaaaa-aabsq-cai
+```
+
+```
+dfx deps deploy dep_b
+Installing canister: yhgn4-myaaa-aaaaa-aabta-cai (dep_b)
+```
+
+:::info
+`dfx deps deploy` always creates the canister with the anonymous identity so that dependencies and application canisters will have different controllers. It also always installs the canister in "reinstall" mode so that the canister status will be discarded.
+:::
+
+## Interactive example
+
+Now that we've explored the concepts and overview of using the `dfx deps` workflow, let's take a look at using an interactive example to demonstrate the functionality. 
+
+This example project will demonstrate an application canister pulling its dependency from the IC mainnet and integrating with it locally. 
+
+- #### Step 1: First, assure that you have installed the [IC SDK](https://github.com/dfinity/sdk) and [git](https://git-scm.com/downloads).
+
+- #### Step 2: Then, open a terminal window and use `git clone` to download the following repository:
+
+```
+git clone https://github.com/lwshang/pullable.git
+```
+
+- #### Step 3: Open the `dfx.json` file in a text or code editor. Declare the pull dependency in `dfx.json` file with the following configuration:
+
+```
+{
+  "canisters": {
+    "service": {
+      "type": "pull",
+      "id": "ig5e5-aqaaa-aaaan-qdxya-cai"
+    }
+  },
+}
+```
+
+Save the file.
+
+- #### Step 4: Use `dfx deps pull` to pull from the mainnet:
+
+```
+dfx deps pull
+Fetching dependencies of canister ig5e5-aqaaa-aaaan-qdxya-cai...
+Found 1 dependencies:
+ig5e5-aqaaa-aaaan-qdxya-cai
+Pulling canister ig5e5-aqaaa-aaaan-qdxya-cai...
+```
+
+- #### Step 5: Configure the `init` argument:
+
+```
+dfx deps init service --argument 1
+```
+
+- #### Step 6: Deploy on a local replica with the commands:
+
+```
+dfx start --clean --background
+dfx deps deploy
+dfx deploy app
+```
+
+- #### Step 7: Test the app by making the following call:
+  
+```
+dfx canister call app double_service
```

Comment:
a little confusing here. why call `app` and not `service`? Does `app` in turn call `service`? Can we explain what is happening here?

Reply:
Updated this step with clarification from @lwshang via Slack DM conversation.